semantic-core/src/Analysis/Eval.hs

```diff
@@ -50,11 +59,10 @@ eval Analysis{..} eval = \case
       if c' then eval t else eval e
     String s -> string s
     Load p -> eval p >>= asString >> unit -- FIXME: add a load command or something
-    Edge e a -> ref a >>= edge e >> unit
-    Frame -> frame
+    Record fields -> traverse (traverse eval) fields >>= record
     a :. b -> do
       a' <- ref a
-      a' ... eval b
+      a' ... b >>= maybe (freeVariable (show b)) (deref' b)
```

Comment:
Projection simply selects (the address of) a field out of a record, rather than pushing into a frame.